File: CleanAddisBackOfficeApi/src/main/java/com/iyoa/cleanaddis/json/PostJson.java
package com.iyoa.cleanaddis.json;

import java.sql.Date;
import java.util.UUID;

import javax.persistence.Column;
import javax.persistence.Id;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.iyoa.cleanaddis.model.Post.CanBeViewedBy;
import com.iyoa.cleanaddis.model.Post.PostStatus;

import lombok.Data;
@Data
public class PostJson {	
	
	@JsonProperty("username")
	private String username;
	
	@JsonProperty("mediaUuid")
	private UUID mediaUuid;
	
	@JsonProperty("date")
	private Date date;
	
	@JsonProperty("noLike")
	private int noLike;
	
	@JsonProperty("noView")
	private int noView;
	
	@JsonProperty("status")
	private PostStatus status;
	
	@JsonProperty("downloadable")
	private int downloadable;
	
	@JsonProperty("canBeViewedBy")
	private CanBeViewedBy canBeViewedBy;
	
	@JsonProperty("allowToBeUsedForArticle")
	private int allowToBeUsedForArticle;	
}

File: app/src/main/java/com/iyoa/cleanaddis/data/posting/PostJSON.kt
package com.iyoa.cleanaddis.data.posting

import androidx.room.PrimaryKey
import com.iyoa.cleanaddis.data.common.MediaJSON
import com.iyoa.cleanaddis.data.common.MediaUUID
import com.squareup.moshi.Json

data class PostJSON(
    @Json(name="uuid") val  uuid: String,
    @Json(name="username") val username:String,
    @Json(name="noLike") val noLike:Int, @Json(name="uuid") val noView: Int,
    @Json(name="downloadable") val downloadable: Int,
    @Json(name="allowToBeUsedForArticle") val allowToBeUsedForArticle:Int,
    @Json(name="media") var media: MediaJSON,
    @Json(name="canBeViewedBy") val canBeViewedBy:String,
    @Json(name="status") val status: String,
    @Json(name="date") val date:String,
    @Json(name="comment") val comments: List<CommentJSON>){

}
